schema-registry/client/src/main/java/com/hortonworks/registries/schemaregistry/client/SchemaRegistryClient.java

```diff
@@ -614,6 +620,81 @@ private static String encode(String schemaName) {
     }
 
     @Override
+    public void enableSchemaVersion(Long schemaVersionId)
+            throws SchemaNotFoundException, SchemaLifecycleException, IncompatibleSchemaException {
+        try {
+            fetchSchemaVersionState(schemaVersionId, "enable");
+        } catch (SchemaLifecycleException e) {
+            Throwable cause = e.getCause();
+            if(cause != null && cause instanceof IncompatibleSchemaException) {
+                throw (IncompatibleSchemaException) cause;
+            }
+            throw e;
+        }
+    }
+
+    @Override
+    public void disableSchemaVersion(Long schemaVersionId) throws SchemaNotFoundException, SchemaLifecycleException {
+        fetchSchemaVersionState(schemaVersionId, "disable");
+    }
+
+    @Override
+    public void deleteSchemaVersion(Long schemaVersionId) throws SchemaNotFoundException, SchemaLifecycleException {
+        fetchSchemaVersionState(schemaVersionId, "delete");
+    }
+
+    @Override
+    public void archiveSchemaVersion(Long schemaVersionId) throws SchemaNotFoundException, SchemaLifecycleException {
+        fetchSchemaVersionState(schemaVersionId, "archive");
+    }
+
+    @Override
+    public void startSchemaVersionReview(Long schemaVersionId) throws SchemaNotFoundException, SchemaLifecycleException {
+        fetchSchemaVersionState(schemaVersionId, "startReview");
+
+    }
+
+    private boolean fetchSchemaVersionState(Long schemaVersionId,
+                                            String operation) throws SchemaNotFoundException, SchemaLifecycleException {
```

Comment:
IMHO, we do not really need a table as the statemachine is always stored as part of the code. 
States are not customizable. I plan to expose an API for statemachine so that UI can use that to build respective actions. 
You raised a valid concern of having strings as operations in REST APIs. This was done to give better expressive REST APIs. These strings were planned to be taken from the state machine api response. But I agree that it is better to be exposed with ids(which are bytes) instead of strings.